boot/freeldr/freeldr/CMakeLists.txt

```
@@ -1,4 +1,6 @@
 
+add_subdirectory(arch)
```

Just do
```suggestion
add_subdirectory(arch/uefi)
```
if possible, that would remove the need of the lone cmakefiles in arch.